server/zio-http-server/src/main/scala/sttp/tapir/server/ziohttp/ZioHttpServerOptions.scala

```diff
@@ -50,5 +52,15 @@ object ZioHttpServerOptions {
       case Some(ex) => ZIO.logDebugCause(msg, Cause.fail(ex))
     }
 
-  def default[R]: ZioHttpServerOptions[R] = customiseInterceptors.options
+  def default[R]: ZioHttpServerOptions[R] = {
+    val decodeFailureHandlerBadRequestOnPathFailure: DecodeFailureHandler =
+      DefaultDecodeFailureHandler.default.copy(
+        respond = DefaultDecodeFailureHandler.respond(
+          _,
+          badRequestOnPathErrorIfPathShapeMatches = true,
+          badRequestOnPathInvalidIfPathShapeMatches = true
+        )
+      )
+    customiseInterceptors[R].copy(decodeFailureHandler = decodeFailureHandlerBadRequestOnPathFailure).options
```

Comment:
let's change this when creating `CustomiseInterceptors` in line 28. That way people will still have the option to override the `DFH` with their own one. We should also probably add some scaladocs there that explain how a custom `DefaultDFH` should be customised.